### PR TITLE
[14.0][MIG] report_qweb_signer

### PR DESCRIPTION
**Pull request by omar7r**

### CODE REVIEW

**Review by chienandalu — COMMENTED** [review]
Hi @omar7r please include #542

**Comment by pedrobaeza** [comment]
There has been also some moves in v13 version to add (and planning to switch to) a new Python library.

**Comment by chienandalu** [comment]
> There has been also some moves in v13 version to add (and planning to switch to) a new Python library.

At the time I didn't find a library mature enough to stick to, although the current implementation isn't very neat anyway...

**Comment by pedrobaeza** [comment]
Please see #528. It seems this one is promising. @sbidoul how were your tests? Should we remove java library in favor of this one?

**Comment by sbidoul** [comment]
@pedrobaeza @chienandalu We have positive results so far with endesive but no large scale production feedback yet. So its probably premature to drop the java code yet.

**Comment by chienandalu** [comment]
Oh, yeah. That the one I had spotted but two years ago it was fairly new

**Comment by pedrobaeza** [comment]
Ok, then at least this PR should be updated with latest 13.0 commits

**Comment by sbidoul** [comment]
And there is also #533 that is pending.

**Comment by pedrobaeza** [comment]
@omar7r did you check about the new sign method? Travis is red by the way.

**Comment by omar7r** [comment]
Yes, I tested two options, both works fine. I think that Travis fails because needs to have installed the package swig in the system, where can I add it?

**Comment by pedrobaeza** [comment]
You can put it on `requirements.txt` file in the root folder.

**Comment by pedrobaeza** [comment]
Should we remove Java library then and leave only the Pythonic one?

**Comment by chienandalu** [comment]
Out of curiosity, how do both methods compare in performance when printing a big number of signed documents?

**Comment by omar7r** [comment]
Swig it's a dependency in the machine not at python level.

In our case, we are using java's library in production, because the project was a migration, from 10.0 to 14.0, I don't test the python library printing several invoices at time. I don't know if someone use the pythonic lib in production and could give feedback, if not, it will be a good test to do, before the decision.

**Comment by pedrobaeza** [comment]
Then you need to change .travis.yml to install such dependency:

https://github.com/OCA/reporting-engine/blob/13.0/.travis.yml#L15

File: report_qweb_signer/models/ir_actions_report.py [hunk 1]
```diff
@@ -39,9 +38,14 @@ def _certificate_get(self, res_ids):
         """Obtain the proper certificate for the report and the conditions."""
         if self.report_type != "qweb-pdf":
             return False
+        company_id = self.env.user.company_id.id
```
Comment (ernestotejeda) [inline report_qweb_signer/models/ir_actions_report.py]: Shouldn't be this self.env.company ?

**Review by ernestotejeda — COMMENTED** [review]
On runbot:
Peek 07-02-2022 17-09

**Comment by ernestotejeda** [comment]
Hello @omar7r , are you working on this PR? If not, I can supersede it to speed up the migration, thanks.

**Comment by omar7r** [comment]
Hi @ernestotejeda 

Yes, but I didn't have much time since your comment, on friday I think I can review it.

**Comment by omar7r** [comment]
@ernestotejeda I tested it again locally and it works, maybe Runbot doesn't have enough memory for running Java.

File: .travis.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ addons:
     packages:
       - expect-dev # provides unbuffer utility
       - graphviz
+      - "swig"
```
Comment (chienandalu) [inline .travis.yml:16]: Mm, why quoted?

Comment (omar7r) [reply .travis.yml:16]: It is like this in 13.0 version

**Review by chienandalu — APPROVED** [review]
Ok, perfect :) Can you squash the migration commits?

**Review by ernestotejeda — APPROVED** [review]
tested locally 👍🏼

**Pull request merged**